### PR TITLE
New module: Jenkins node management

### PR DESCRIPTION
**Pull request by phyrwork**
##### SUMMARY
A while back l proposed PR https://github.com/ansible-collections/community.general/pull/7160 for new module: Jenkins node management.

It was closed before I could complete it - reviewer was largely satisfied but I didn't want to submit without completing the unit tests.

This PR is a scaled back version of the original - fewer features (to be added incrementally later) but unit tests completed.

Implemented and tested:

- Present, absent, enabled, disabled Jenkins node.
- Set node # executors.
- Set node labels.

Tested against latest Jenkins LTS at time of writing.

Included workaround for `python-jerkins` redirect authentication error present in some versions of this dependency.

##### ISSUE TYPE
- New Module/Plugin Pull Request

##### COMPONENT NAME
jenkins_node

### CODE REVIEW

File: plugins/modules/jenkins_node.py [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+
+
+IS_PYTHON_2 = sys.version_info[0] <= 2
+
+
+class JenkinsNode:
+    def __init__(self, module):
+        self.module = module
+
+        self.name = module.params.get('name')
```
Comment (russoz) [inline plugins/modules/jenkins_node.py:161]: Nothing *wrong* with this, but the use of `.get()` here is a bit redundant: `module.params` will **always** contain all the parameters, even when their value is `None`.

File: plugins/modules/jenkins_node.py [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+            user=dict(),
+            token=dict(no_log=True),
+            state=dict(choices=['enabled', 'disabled', 'present', 'absent'], default='present'),
+            num_executors=dict(type='int'),
+            labels=dict(type='list', elements='str'),
+        ),
+        supports_check_mode=True,
+    )
+
+    test_dependencies(module)
```
Comment (russoz) [inline plugins/modules/jenkins_node.py:376]: You might want to consider using `module_utils.deps`, see https://docs.ansible.com/ansible/latest/collections/community/general/docsite/guide_deps.html for guidance on how to use it.

File: plugins/modules/jenkins_node.py [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: jenkins_node
+short_description: Manage Jenkins nodes
+description:
```
Comment (russoz) [inline plugins/modules/jenkins_node.py:15]: This is mandatory:
```suggestion
version_added: 10.0.0
description:
```

**Review by russoz — COMMENTED** [review]
Hi @phyrwork , thanks for your contribution!

I have commented on a couple of minor issues in docs, and a couple of suggestions to the code. Other than that, it looks good.

**Comment by phyrwork** [comment]
I think I have resolved all suggestions - let me know if further changes are required.

**Review by russoz — APPROVED** [review]
Yup, @phyrwork thanks for that.

LGTM

**Comment by felixfontein** [comment]
@phyrwork thanks for your contribution!
@russoz thanks for reviewing!

**Pull request merged**